pkg/schedulingtypes/replicascheduler.go

```diff
@@ -48,43 +48,43 @@ func init() {
 	RegisterSchedulingType(RSPKind, NewReplicaScheduler)
 }
 
+type pluginArgs struct {
+	kubeClient             kubeclientset.Interface
+	crClient               crclientset.Interface
+	fedNamespace           string
+	clusterNamespace       string
+	targetNamespace        string
+	federationEventHandler func(pkgruntime.Object)
+	clusterEventHandler    func(pkgruntime.Object)
+	handlers               *ClusterLifecycleHandlerFuncs
+}
+
 type ReplicaScheduler struct {
-	plugins         map[string]*Plugin
+	plugins map[string]*Plugin
+	pluginArgs
+
 	fedClient       fedclientset.Interface
 	podInformer     FederatedInformer
 	targetNamespace string
 }
 
 func NewReplicaScheduler(fedClient fedclientset.Interface, kubeClient kubeclientset.Interface, crClient crclientset.Interface, fedNamespace, clusterNamespace, targetNamespace string, federationEventHandler, clusterEventHandler func(pkgruntime.Object), handlers *ClusterLifecycleHandlerFuncs) Scheduler {
```

Comment:
(No action required) Consider having this method accept the `pluginArgs` struct.